File: src/App/UserSettings.js
Bridge.assembly("Widgetoko", function ($asm, globals) {
    "use strict";

    Bridge.define("Widgetoko.MainProcess.UserSettings", {
        $literal: true,
        statics: {
            methods: {
                Deserialize: function (data) {
                    var deobfuscatedDataArr = System.Convert.fromBase64String(data);
                    var serializedData = System.Text.Encoding.Default.GetString(deobfuscatedDataArr);
                    var obj = Newtonsoft.Json.JsonConvert.DeserializeObject(serializedData, System.Object);

                    return obj;
                }
            }
        },
        methods: {
            Serialize: function () {
                var data = Newtonsoft.Json.JsonConvert.SerializeObject(this);
                var dataArr = System.Text.Encoding.Default.GetBytes$2(data);
                var obfuscatedData = System.Convert.toBase64String(dataArr, null, null, null);

                return obfuscatedData;
            }
        }
    });
});

File: Widgetoko/MainProcess/UserSettings.cs
using System;
using System.Text;
using Bridge;
using Newtonsoft.Json;
using Widgetoko.Twitter;

namespace Widgetoko.MainProcess
{
    [ObjectLiteral]
    public class UserSettings
    {
        public TwitterCredentials Credentials { get; set; }

        public string Serialize()
        {
            var data = JsonConvert.SerializeObject(this);
            var dataArr = Encoding.Default.GetBytes(data);
            var obfuscatedData = Convert.ToBase64String(dataArr);

            return obfuscatedData;
        }

        public static UserSettings Deserialize(string data)
        {
            var deobfuscatedDataArr = Convert.FromBase64String(data);
            var serializedData = Encoding.Default.GetString(deobfuscatedDataArr);
            var obj = JsonConvert.DeserializeObject<UserSettings>(serializedData);

            return obj;
        }
    }
}